openghg/util/_domain.py

```diff
@@ -118,24 +118,135 @@ def _get_coord_data(coord: str, data: Dict[str, Any], domain: str) -> ndarray:
     return coord_data
 
 
-def convert_longitude(
-    longitude: ndarray, return_index: bool = False
-) -> Union[ndarray, Tuple[ndarray, ndarray]]:
-    """Convert longitude extent to -180 - 180 and reorder.
+def find_coord_name(data: XrDataLike, options: List[str]) -> Optional[str]:
+    """
+    Find the name of a coordinate based on input options.
+    Only the first found value will be returned.
+
+    Args:
+        data: xarray Data structure
+        options: List of options to check. Will be checked in order.
+
+    Returns:
+        str / None: Name of coordinate if located within data. None otherwise.
+    """
+    for option in options:
+        if option in data.coords:
+            name = option
+            break
+    else:
+        return None
+
+    return name
+
+
+def convert_longitude(longitude: ArrayLikeMatch) -> ArrayLikeMatch:
+    """
+    Convert longitude extent from (0 to 360) to (-180 to 180).
+    This does *not* reorder the values.
 
     Args:
-        longitude: Array of valid longitude values in degrees.
-        return_index: Return re-ordering index as well as updated longitude
+        longitude: Valid longitude values in degrees.
     Returns:
-        ndarray(, ndarray) : Updated longitude values and new indices if requested.
+        ndarray / DataArray : Updated longitude values in the same order.
     """
     # Check range of longitude values and convert to -180 - +180
-    mtohe = longitude > 180
-    longitude[mtohe] = longitude[mtohe] - 360
-    ordinds = np.argsort(longitude)
-    longitude = longitude[ordinds]
+    longitude = ((longitude - 180) % 360) - 180
+
+    return longitude
+
+
+def convert_internal_longitude(data: XrDataLikeMatch,
+                               lon_name: Optional[str] = None,
+                               reorder: bool = True) -> XrDataLikeMatch:
+    """
+    Convert longitude coordinate within an xarray data structure (DataArray or Dataset).
+
+    Args:
+        data: Data with longitude values to convert.
+        lon_name: By default will look a coord called "lon" or "longitude".
+            Otherwise must be specified.
+        reorder: 
```

Comment:
This docstring needs finishing.

Reply:
Good spot - I'll take a look at that now.

Reply:
This has been updated